Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -115,6 +115,12 @@ private string IonFieldNameFromProperty(PropertyInfo property)
             {
                 return ((IonPropertyName)ionPropertyName).Name;
             }
+
+            if (options.PropertyNameCaseInsensitive)
+            {
+                return property.Name;
```

Comment:
I'm not 100% sure why this is needed for case insensitivity. Is this correct?

Reply:
The idea is to just return the property name as-is when it's case insensitive rather than bothering with the naming convention logic below (which handles conversions to camel case, snake case, etc).

Reply:
On 2nd thought.....snake case involves adding underscores so we should still do the naming convention conversion regardless of case sensitivity. I have now removed this piece of logic.